lib/addons/core-addon/index.ts

```diff
@@ -20,11 +40,25 @@ export class CoreAddOn implements ClusterAddOn {
     }
 
     deploy(clusterInfo: ClusterInfo): void {
+
+        // Create a service account if user provides namespace and service account
+        let serviceAccountRoleArn: string | undefined = undefined;
+        let namespace: string = this.coreAddOnProps?.namespace ? this.coreAddOnProps.namespace : DEFAULT_NAMESPACE;
+
+        if (this.coreAddOnProps?.policyDocument) {
+            const serviceAccount = createServiceAccount(clusterInfo.cluster, this.coreAddOnProps.addOnName,
+                namespace, this.coreAddOnProps.policyDocument);
+            serviceAccountRoleArn = serviceAccount?.role?.roleArn;
```

Comment:
I think this should be `serviceAccountRoleArn = serviceAccount.role.roleArn;`  as `createServiceAccount()` is guaranteed to return an SA.

Reply:
Updated. Thanks!